app/models/assignee.rb

```diff
@@ -0,0 +1,4 @@
+class Assignee < ApplicationRecord
```

Comment:
why so complex ...

rework it. remove this association.

Task has only one assignee. So task `belongs_to :assignee, class_name: "User"`